frame/support/src/storage/generator/double_map.rs

```diff
@@ -371,17 +380,30 @@ impl<
 		}
 	}
 
+	fn iter_key_prefix_from(
+		k1: impl EncodeLike<K1>,
+		starting_raw_key: Vec<u8>,
+	) -> Self::PartialKeyIterator {
+		let mut iter = Self::iter_key_prefix(k1);
+		iter.set_last_raw_key(starting_raw_key);
+		iter
+	}
+
 	fn drain_prefix(k1: impl EncodeLike<K1>) -> Self::PrefixIterator {
 		let mut iterator = Self::iter_prefix(k1);
 		iterator.drain = true;
 		iterator
 	}
 
 	fn iter() -> Self::Iterator {
+		Self::iter_from(G::prefix_hash())
```

Comment:
Wee bit strange here. `iter_from` is already reading `let prefix = G::prefix_hash();`, but we pass in `G::prefix_hash()` to it again. I'd make the `starting_raw_key` an `Option`.

Reply:
I guess we can just revert this and implement `iter_from` in terms of `iter` by calling `set_previous_key` on the resulting `PrefixIterator`.